topology.go

```diff
@@ -369,8 +372,11 @@ func (t *Topology) runFilterChain() {
 			// Validate against patterns
 			if t.validate != nil {
 				// call external validation function
-				if ok, idx := t.validate(record); !ok {
-					atomic.AddInt64(&t.invalid[idx], 1)
+				ok, idx := t.validate(record)
+				if !ok {
```

Comment:
is it as fast as the previous version?

Reply:
No it's not since incrementing an atomic is faster than locking a mutex. However this is the only correct way to do it. Moreover the change is only seen when there's an invalid logline, the happy path stays the same